PACKAGE = "caato2_reverse_docker"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("AnglePID_dt", double_t, 0, "dt, the time interval to calculate with", 0.1, 0.01, 1)
gen.add("AnglePID_max_angular_vel", double_t, 0, "mav_vel, the maximum Angular velocity of the robot", 0.3, 0.0, 0.5)
gen.add("AnglePID_min_angular_vel", double_t, 0, "min_vel, the minimum Angular velocity of the robot", 0.0, 0.0, 0.1)
gen.add("AnglePID_Kp", double_t, 0, "Kp, the Proportional gain for the PID Angle Controller", 0.5, 0.5, 1.0)
gen.add("AnglePID_Ki", double_t, 0, "Ki, the Integral gain for the PID Angle Controller", 0.0, 0.0, 1.0)
gen.add("AnglePID_Kd", double_t, 0, "Kd, the Differential gain for the PID Angle Controller", 0.0, 0.0, 1.0)

gen.add("XVelPID_dt", double_t, 0, "dt, the time interval to calculate with", 0.1, 0.01, 1)
gen.add("XVelPID_max_x_vel", double_t, 0, "mav_vel, the maximum X velocity of the robot", -0.5, -1.0, -0.1)
gen.add("XVelPID_min_x_vel", double_t, 0, "min_vel, the minimum X velocity of the robot", 0.0, -0.1, 0.0)
gen.add("XVelPID_Kp", double_t, 0, "Kp, the Proportional gain for the PID Vel Controller", 0.0, 0.0, 1.0)
gen.add("XVelPID_Ki", double_t, 0, "Ki, the Integral gain for the PID Vel Controller", 0.0, 0.0, 1.0)
gen.add("XVelPID_Kd", double_t, 0, "Kd, the Differential gain for the PID Vel Controller", 0.0, 0.0, 1.0)

exit(gen.generate(PACKAGE, "reverse_docker", "PID"))